docs/contributing/contributing-rules.md

```diff
@@ -1,151 +1,415 @@
+Fork our repository and make a pull request; we'll contact you about signing our Contributor License Agreement (CLA). Install pre-commit (see [installing pre-commit](#installing-pre-commit)) and make a pull request to the [Semgrep Registry](https://github.com/returntocorp/semgrep-rules) with two files:
 1. The semgrep pattern (as YAML file).
 2. The test file (with the file extension of the language or framework). The test file must contain at least one true positive and one true negative test case to be approved. See the [Tests](#tests) section of this document for more information.
 
-See an example of a [pull request](https://github.com/returntocorp/semgrep-rules/pull/1728/files) to the rules repository. Pull requests require the approval of at least one maintainer and successfully passed [CI jobs](https://github.com/returntocorp/semgrep-rules/actions).
+See an example of a [pull request](https://github.com/returntocorp/semgrep-rules/pull/1728/files) to the Semgrep Registry. Pull requests require the approval of at least one maintainer and successfully passed [CI jobs](https://github.com/returntocorp/semgrep-rules/actions).
 
-Find more about the rules repository by reading the [Rule writing](#rule-writing) and [Tests](#tests) sections.
+Find more about the Semgrep Registry by reading the [Rule writing](#rule-writing) and [Tests](#tests) sections.
 
-#### Installing `pre-commit`
+## Rule requirements
 
-If you are contributing to the rules repository through GitHub but you are **not** contributing through sharing your rule in Semgrep App, install `pre-commit`. Follow [pre-commit installation documentation](https://pre-commit.com/#installation). Once `pre-commit` is set up you may commit code and create pull requests to rules repository.
+Every rule submitted to Semgrep Registry must include:
 
-## Rule writing
+<dl>
+    <dt>Rule ID</dt>
+    <dd>Rule ID is created automatically by Semgrep App, but include it if you are contributing solely through GitHub without Semgrep App. Use a meaningful name for the rule ID. Example: <code>- id: detect-angular-sce-disabled</code>.
+    </dd>
+    <dt>Test file</dt>
+    <dd>Test file in the language that the rule is targeting. See <a href="#tests"> Tests</a> for more details.</dd>
+    <dt>Rule message</dt>
+    <dd>Include details about the matched pattern and inform about mitigation of any related issues. See <a href="#rule-messages"> Rule messages</a> for more details.</dd>
+    <dt>Languages</dt>
+    <dd>Specify for which languages or language is the rule meant. For example:
+    <pre>
+    languages:<br />
+    - javascript<br />
+    - typescript
+    </pre>
+    </dd>
+    <dt>Severity</dt>
+    <dd>TODO - Description of severity is. Possible values: <code>INFO</code>, <code>WARNING</code>, <code>ERROR</code>, <code>INVENTORY</code></dd>
```

Comment:
One of the last tasks here. :) Please @LewisArdern, describe what severity is and what values should be used... I checked semgrep-rules codebase to see some, but I am not sure what exactly these values mean (INFO, WARNING, ERROR, INVENTORY) - is inventory intended for end users?